docs/cmd/kn_service_create.md

```diff
@@ -45,7 +45,7 @@ kn service create NAME --image IMAGE [flags]
 ### Options
 
 ```
-      --annotation stringArray       Service annotation to set. name=value; you may provide this flag any number of times to set multiple annotations. To unset, specify the annotation name followed by a "-" (e.g., name-).
+  -a, --annotation stringArray       Service annotation to set. name=value; you may provide this flag any number of times to set multiple annotations. To unset, specify the annotation name followed by a "-" (e.g., name-).
```

Comment:
Im fine with that (except maybe that it can be easily confused with `-A` which is "for all namespaces")

Reply:
on the other hand, why not `-a` for `--arg` ? 

But as we already have `-e` and `-l`, for consistencies sake I'm fine with `-a` (but in general we should take care to give away single letter options not easily)

Reply:
I think --annotation would be more than --arg, plus "arg" is shorter than "annotation" - so between the two I don't think "arg" needs as much UX help :-)

As for -A and -a .... not sure what to say, other than these things are case sensitive for a reason.